### ISSUE TITLE
Replace the lists of variables in HttpConstants with enum class

### ISSUE THREAD

**Issue by Breigner01** [martin-olivier/ZiAPI#14]
### Description

The lists of variables that are used as number can and should be replaced by enum class.

### Is your feature request related to a problem? Please describe.

Nope

### Describe the solution you'd like

Replace the lists that can be by enum class

### Expected behaviour

Same

### Actual behaviour

Same

### Logs

None

### Additional context

None

**Comment by rojasdiegopro** [comment]
Could you provide justification on why this is a good idea? :)

**Comment by Breigner01** [comment]
Enum is made for lists of integers replaced by names, where as `constexpr` can be used for any type so when you have a list of numbers used the way it is now, the standard way would want you to create at least an enum.
Since it's C++11 and higher an enum class should be used so that we keep the `my_enum::my_value` which is similar to the one you had: `my_namespace::my_variable`

It shouldn't be much different on the compiled side since enum and enum class are defined using constexpr see [Enum](https://en.cppreference.com/w/cpp/language/enum)

**Comment by Ph4ntomas** [comment]
In general, I think you should always try to use named types whenever possible, as they usually enforce the production of cleaner and safer code.

For example, currently, a function that takes an HTTP code  as it's parameter expect an `int` (wich in itself is unsafe, but more on that later), so I could just ignore these constant, input the value myself, and make a typo or something. With an enum class, I can input my own codes, but I'd have to explicitly cast to the named type, making the code safer and more explicit overall.

> For example I think an enum class should be used if you have the following
>
> ```c++
> enum class PlayerState {
>     Crouching,
>     Walking,
>     Running,
>     Idle
> };
> ```
> Where the value don't have any numeric values.

I don't really agree with that statement. Enum are just a way to define a named numerical value with restricted "standard" values.
The fact that sometime, you don't care about the underlying value doesn't mean it doesn't exist, and it's quite common to specify the value when it need to have one.

> The reason why I chose not to go with an enum in this one is because I use constexpr auto when I'm dealing with a value that has a meaningful numeric value.

Imo it's not a good practice to do this, as you don't know the deduced type.
For example, one could write 
```c++
    constexpr auto CONTINUE = 100;
    constexpr auto MY_SPECIAL_CODE = 2147483648
```
In this example, CONTINUE is deduced as an `int`, but MY_SPECIAL_CODE is deduced as a `long` or a `long long`, which would introduce a narrowing conversion if a function is expecting an `int`. 
Whenever possible, you should properly specify the type of your constant to prevent this.

With an enum however, that example might change the underlying value of the enum, but the change will be consistent across every values.

I'm not against using the `auto` keyword, but I think you should never use it in the global scope, or anywhere it would create ambiguities, that could potentially creates bugs later on, especially if it's not strictly needed.

**Comment by rojasdiegopro** [comment]
Okay! Understood, we will be shipping enum types in place of constant expressions for numerical http constants in the next version of the API (2.2.0).
Thanks @Breigner01 for the issue and the discussion.

**Comment by rojasdiegopro** [comment]
Fixed with release 2.2.0
